### PR TITLE
RESOURCE-139 Support adls paths

### LINKED ISSUE #483: Data Lake Storage Gen 2 Path

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

Support ADLS Path(s)

### Issues Resolved
#483 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]
LGTM.

**Comment by IanMadd** [comment]
It looks like a bunch of files from #482 got pulled into this PR.

File: docs/resources/azure_data_lake_storage_gen2_path.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). For an example, `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+`name`, `account_name` and `path` is a required parameter and `dns_suffix` is an optional parameter.
+
+```ruby
+describe azure_data_lake_storage_gen2_path(account_name: 'ACCOUNT_NAME', filesystem: 'FILE_SYSTEM', name: 'PATHNAME') do
```
Comment (IanMadd) [inline docs/resources/azure_data_lake_storage_gen2_path.md:28]: There's a mismatch between the parameters here and ones listed in below in the parameters table.

**Comment by sathish-progress** [comment]
> It looks like a bunch of files from #482 got pulled into this PR.

yeah this PR has core dependencies on #482, we will first merge #482 and than merge this so we will see changes only for this PR

**Comment by clintoncwolfe** [comment]
@sathish-progress can you please clean up this PR and resolve the conflicts?

File: docs/resources/azure_data_lake_storage_gen2_path.md (hunk 1 above)
Comment (sathish-progress) [reply docs/resources/azure_data_lake_storage_gen2_path.md]: updated docs!

**Review by IanMadd — APPROVED** [review]

**Pull request merged**